### PR TITLE
Add regularized coulomb & debris friction tests for MISMIP+

### PR DESCRIPTION
**Pull request by matthewhoffman**
This PR adds new MALI tests for the new debris friction law capability and the existing regularized coulomb capability.  It adds variants to the MISMIP+ smoke test for these friction laws and the existing weertman friction law.   Each of three variants have separate yaml files appropriate for that friction law.  The standard input file consisting of a spunup 2km state is modified for the regularized coulomb and debris friction variants to use appropriate parameters values as input fields.  These are not meant to be consistent with the spunup glacier state in the file, but allow the smoke test to test each of the 3 friction laws for a reasonable glacier state.  The PR also adds the RC and DF variants to the full_integration suite so that we have test coverage for those friction laws. 

Checklist
* [x] User's Guide has been updated
* [x] Developer's Guide has been updated
* [x] API documentation in the Developer's Guide (`api.rst`) has any new or modified class, method and/or functions listed
* [x] Documentation has been [built locally](https://mpas-dev.github.io/compass/latest/developers_guide/building_docs.html) and changes look as expected
* [x] The `MALI-Dev` submodule has been updated with relevant MALI changes
* [x] Document (in a comment titled `Testing` in this PR) any testing that was used to verify the changes
* [x] New tests have been added to a test suite

### CODE REVIEW

**Review by mperego — APPROVED** [review]
Looks good to me.

File: compass/landice/tests/mismipplus/albany_input_debrisfriction.yaml [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+          Sliding Velocity Regularization: 5.0e2
+          Flow Rate Type: Constant
+          Flow Rate: 6.338e-25      # Pa^-n s^-1
+  # # Discretization Description
+  # Discretization:
+  #   Exodus Output File Name: albany_output.exo
+  #   Side Set Discretizations:
+  #     basalside:
+  #       Exodus Output File Name: albany_output_basal.exo
+
```
[the remark this answers is not in the document]
Comment (mperego) [reply compass/landice/tests/mismipplus/albany_input_debrisfriction.yaml]: > Are the flow rate lines supposed to be in their own section `LandIce Viscosity` , or is it okay to set them there under `Basal Friction Coefficient`?

The flow-rate lines in the `Basal Friction Coefficient` section are only used for the boundary condition. If they are not specified in the `Basal Friction Coefficient` section, they are inferred from the ones in the `LandIce Viscosity`.

File: compass/landice/tests/mismipplus/run_model.py [hunk 2]
```diff
@@ -184,3 +238,23 @@ def run(self):
         for suffix in self.suffixes:
             run_model(step=self, namelist='namelist.{}'.format(suffix),
                       streams='streams.{}'.format(suffix))
+
+    def _update_mesh_for_debris_friction(self):
+        mesh_filename = os.path.join(self.work_dir, self.mesh_file)
+
+        with xr.open_dataset(mesh_filename) as ds_mesh:
+            ds_mesh['muFriction'].loc[:] = 0.4
+            ds_mesh['bedRoughnessBC'] = xr.full_like(ds_mesh['muFriction'],
```
[the remark this answers is not in the document]
Comment (matthewhoffman) [reply compass/landice/tests/mismipplus/run_model.py]: Good catch!

**Comment by matthewhoffman** [comment]
# Testing

Testing with landice full_integration suite described here: https://github.com/MALI-Dev/E3SM/pull/135#issuecomment-5542136252

All tests pass, including the newly added RC and DF smoke tests.
